sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -159,9 +159,11 @@ class DataFrame protected[sql](
 
   /**
    * Internal API for Python
+   * @param numRows Number of rows to show
    */
-  private[sql] def showString(): String = {
-    val data = take(20)
+  private[sql] def showString(numRows: Int = 20): String = {
```

Comment:
there is no reason to set a default param value, just have showString(numRows: Int).